File: Core/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace PriceAdvisor.Core.Models
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Code { get; set; }
        [DefaultValue(false)]
        public bool Edited {get;set;}
        public ICollection<Price> Prices { get; set; }
    }
}

File: ClientApp/app/models/product.ts
import { Price, SavePrice } from './price';
import { Eshop } from './eshop';

export interface Product {
    id: number; 
    name: string;
    code: string; 
    edited: boolean;   
    prices: Price;
    eshops: Eshop;
  }

  export interface SaveProduct {
    id: number; 
    code: string;
    name: string;   
  }